website/src/docs/hotchocolate/performance/persisted-queries.md

```diff
@@ -2,12 +2,239 @@
+```
+
+## Step 2: Configure persisted queries on server
+
+Next, we want to configure our GraphQL server to be able to handle persisted query requests. For this, we need to register the corresponding query storage and configure the persisted query request pipeline.
+
+1. Configure GraphQL server to use the persisted query pipeline.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddRouting()
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .UsePersistedQueryPipeline();
+}
+```
+
+2. Next, register the file-query storage. (You can use any other, this example use file-storage).
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddRouting()
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .AddReadOnlyFileSystemQueryStorage("./persisted_queries");
+}
+```
+
+By this HC will search for correspondent query Hash in folder `./persisted_queries`. It is important to store all exported clients persisted  queries in this folder using a specific name and format: `{Hash}.graphql`
+
+Example: `0c95d31ca29272475bf837f944f4e513.graphql`
+
+Now your server knows where to search for the requested query `Id` and loads it in memory in case of a successful match. 
+
+
+## Step 3: Configure persisted  queries on the client (Relay)
+
+This step shows what you need to configure on the client-side or adjust on the server regarding clients. Example show Relay implementation. You can use any other GraphQL client like Apollo. The configuration is always client-related so follow offical client Docs. 
+
+[Docs Relay](https://relay.dev/docs/guides/persisted-queries/)
+
+[Docs Apollo](https://www.apollographql.com/docs/apollo-server/performance/apq/)
+
+### What can be different between clients
+
+- Hashing algorithm - Hot Chocolate server is configured to use by default the MD5 hashing algorithm. HotChocolate server comes out of the box with support for MD5, SHA1, and SHA256
+
+- Default Headers and Setup - HC expect the persisted `Id` to be named as` id` in the request header. Some docs like Relay show you to send it under `doc_id` this would not work with HC. You must adjust the client fetch function to use the correct variable. 
+
+### Hashing algorithm setup
+
+If adjustmed of algorithm is required you can do it using following sintax:
+
+ > **ℹ️**  For relay it is not needed. Relay use default MD5 Hash.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        // Global Services
+        .AddRouting()
+        .AddMemoryCache()
+        .AddSha256DocumentHashProvider(HashFormat.Hex)
+        
+        // GraphQL server configuration
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .UsePersistedQueryPipeline()
+        .AddReadOnlyFileSystemQueryStorage("./persisted_queries");
+}
+```
+
+### Relay persisted queries setup
```

Comment:
I don't think this belongs here. This is already covered in detail and kept up-to-date in the [official documentation](https://relay.dev/docs/guides/persisted-queries/). You've already linked to it above, so I would just get rid of it here.

Reply:
The problem is official documentation does things differently. For example, use `doc_id` which is not compatible with what HC requires. HC requires this variable to be named as `id`... From my perspective, it is better to understand and have it in one place with an explanation of why that why I put (just my idea on time when I wrote)... Also, the end of Docs is specific.. The relay output is not compatible with HC so yo cannot make it work with Relay or just HC docs...

Reply:
Ah, I see. I would still get rid of the part with the packages and the general explanation, but the later part is of course justified then!